File: 0455-assign-cookies/0455-assign-cookies.ts
function findContentChildren(g: number[], s: number[]): number {
    g.sort((a, b) => a - b);
    s.sort((a, b) => a - b);

    let p = 0, i = 0;

    while (p < g.length && i < s.length) {
        if (g[p] <= s[i]) {
            p++;
        }
        i++;
    }

    return p;
};

File: 0455-assign-cookies/0455-assign-cookies.rs
impl Solution {
    pub fn find_content_children(mut g: Vec<i32>, mut s: Vec<i32>) -> i32 {
        g.sort(); // Sort greed factors
        s.sort(); // Sort cookie sizes

        let (mut p, mut i) = (0, 0); // p -> children, i -> cookies

        while p < g.len() && i < s.len() {
            if g[p] <= s[i] {
                p += 1; // Move to next child
            }
            i += 1; // Move to next cookie
        }

        p as i32 // Number of content children
    }
}
